src/metatrain/experimental/nanopet/modules/samples.py

```diff
@@ -0,0 +1,535 @@
+
+    If ``include_atom_types=False``, the edge samples values with dimensions
+    corresponding to ["system", "first_atom", "second_atom", "cell_shift_a",
+    "cell_shift_b", "cell_shift_c"] are returned.
+
+    If ``include_atom_types=True``, the edge samples values with the dimensions as above
+    but with prepended dimensions ["first_atom_type", "second_atom_type"] are returned.
+    """
+    # Use each system's neighbor list to get the indices
+    edge_sample_values_: List[torch.Tensor] = []
+    for system_id, system in enumerate(systems):
+        sample_values_edge_system = system.get_neighbor_list(
+            system.known_neighbor_lists()[0]
+        ).samples.values
+        system_id_ = (
+            torch.ones(sample_values_edge_system.shape[0], device=system.device)
+            * system_id
+        )
+        edge_sample_values_.append(
+            torch.cat(
+                (system_id_.unsqueeze(1), sample_values_edge_system),
+                dim=1,
+            )
+        )
+    edge_sample_values = torch.vstack(edge_sample_values_).to(dtype=torch.int32)
+
+    if not include_atom_types:
+        return edge_sample_values
+
+    # For the systems, get the atomic types
+    atomic_types: Dict[int, torch.Tensor] = {}
+    for system_i, system in enumerate(systems):
+        atomic_types[system_i] = system.types
+
+    # Insert the atom type dimensions
+    edge_first_atom_type = torch.tensor(
+        [
+            atomic_types[system_i.item()][atom_i]
+            for system_i, atom_i in zip(
+                edge_sample_values[:, 0],
+                edge_sample_values[:, 1],
+            )
+        ],
+        dtype=torch.int32,
+    ).reshape(-1, 1)
+    edge_second_atom_type = torch.tensor(
+        [
+            atomic_types[system_i.item()][atom_i]
+            for system_i, atom_i in zip(
+                edge_sample_values[:, 0],
+                edge_sample_values[:, 2],
+            )
+        ],
+        dtype=torch.int32,
+    ).reshape(-1, 1)
+
+    return torch.hstack(
+        [
+            edge_first_atom_type,
+            edge_second_atom_type,
+            edge_sample_values,
+        ]
+    )
+
+
+def symmetrize_edge_samples(
+    edge_sample_values: torch.Tensor,
+) -> torch.Tensor:
+    """Symmetrizes the samples labels for the PET edge features."""
+
+    # 1) first_atom_type < second_atom_type: no symmetrization required. Ensure
+    #    triangular in atom type.
+    diff_atom_type = edge_sample_values[:, 0] < edge_sample_values[:, 1]
+    edge_sample_values_diff_types = edge_sample_values[diff_atom_type]
+    edge_sample_values_diff_types = torch.hstack(
+        [
+            torch.zeros(len(edge_sample_values_diff_types), dtype=torch.int32).reshape(
+                -1, 1
+            ),  # s2_pi = 0
+            edge_sample_values_diff_types,
+        ]
+    )
+
+    # 2) first_atom_type == second_atom_type: symmetrization required. Ensure triangular
```

Comment:
maybe explain (here and in other places where you refer to it) what "triangular" means?